liquidator/OneInchExchange.js

```diff
@@ -0,0 +1,73 @@
+const Token = artifacts.require("ExpandedERC20");
+
+const oneSplitAbi = require("./abi/OneSplit.json");
+const oneSplitAddress = "0xC586BeF4a0992C495Cf22e1aeEE4E446CECDee0E";
```

Comment:
Just curious -- what happens if we try to use the liquidator bot on Kovan? Is there a `OneSplit` deployment on Kovan? Is it at the same address?

I think this is something we can figure out later as we try to integrate this into the liquidator bot. One thing we do for other contracts is that we handle this in migrations -- we determine whether there is a real deployment of this contract on this network id. If not, we deploy a mock version. Then the artifact includes the address regardless and the implementation doesn't need to worry about whether it exists or not or how the address is retrieved.

Because we do a lot of work on testnets and Ganache, I think we should generally avoid hardcoding mainnet addresses in code.

No worries for now -- we can figure this out offline. This should work since this is only currently being tested on a mainnet fork.

Reply:
> Just curious -- what happens if we try to use the liquidator bot on Kovan? Is there a OneSplit deployment on Kovan? Is it at the same address?

1Inch is only deployed on mainnet, because its the only network that can guarantee consistent aggregation of DEX protocols (curve isn't on kovan, and its the main stablecoin exchange). Additionally you'll find that DEXes that are on testnets usually don't have interopable ERC20's. For example, DAI in Compound might be different from DAI in Uniswap on Kovan.

> One thing we do for other contracts is that we handle this in migrations -- we determine whether there is a real deployment of this contract on this network id. If not, we deploy a mock version

This might need to be in a sprint of its own.

> Then the artifact includes the address regardless and the implementation doesn't need to worry about whether it exists or not or how the address is retrieved.

Say we would like to keep the integration tests, how would this work for mainnet?

> No worries for now -- we can figure this out offline. This should work since this is only currently being tested on a mainnet fork

:+1:  I'll bring it up in the sprint planning

Reply:
> 1Inch is only deployed on mainnet, because its the only network that can guarantee consistent aggregation of DEX protocols (curve isn't on kovan, and its the main stablecoin exchange). Additionally you'll find that DEXes that are on testnets usually don't have interopable ERC20's. For example, DAI in Compound might be different from DAI in Uniswap on Kovan.

Makes sense. Then we might want to have truffle deploy our own minimal mock on the testnets and on Ganache.

> This might need to be in a sprint of its own.

Mocking 1inch shouldn't be too bad. We can try to tackle this next sprint. Happy to help with this part as it might be kind of opaque how this is done since we don't do it too much. `TestnetERC20` (badly named) is basically our DAI mock and it works similarly.

> Say we would like to keep the integration tests, how would this work for mainnet?

If we set things up properly, since the mainnet fork has the same network ID, our infra should realize that there's a real deployment of 1inch on network id 1 and not try to deploy a mock in the migrations. So the fork should work exactly as expected.

> 👍 I'll bring it up in the sprint planning

Sounds good!